router/batchrouter/asyncdestinationmanager/sftp/utils.go

```diff
@@ -0,0 +1,263 @@
+	}
+	if err := scanner.Err(); err != nil {
+		return nil, err
+	}
+	return records, nil
+}
+
+func generateFile(filePath string, format string, fileName string) (string, error) {
+	switch strings.ToLower(format) {
+	case ".json":
+		return generateJSONFile(filePath, fileName)
+	case ".csv":
+		return generateCSVFile(filePath, fileName)
+	default:
+		return "", errors.New("unsupported format")
+	}
+}
+func generateJSONFile(filePath string, fileName string) (string, error) {
+	// Open the input file
+	file, err := os.Open(filePath)
+	if err != nil {
+		return "", err
+	}
+	defer file.Close()
+
+	var data interface{}
+	err = json.NewDecoder(file).Decode(&data)
+	if err != nil {
+		return "", err
+	}
+
+	// Generate the desired file path
+	desiredFilePath := filepath.Join(os.TempDir(), fileName)
+
+	// Create a temporary CSV file
+	tempFile, err := os.Create(desiredFilePath)
+	if err != nil {
+		return "", err
+	}
+	defer tempFile.Close()
+
+	// Write JSON data to the temporary file
+	encoder := json.NewEncoder(tempFile)
+	encoder.SetIndent("", "  ")
+	err = encoder.Encode(data)
+	if err != nil {
+		return "", err
+	}
+
+	// Get the absolute path of the temporary file
+	absPath, err := filepath.Abs(tempFile.Name())
+	if err != nil {
+		return "", err
+	}
+
+	return absPath, nil
+}
+
+func generateCSVFile(filePath string, fileName string) (string, error) {
+	// Parse JSON records
+	records, err := parseRecords(filePath)
+	if err != nil {
+		return "", err
+	}
+
+	// Extract field names
+	fieldNames, err := getFieldNames(records)
+	if err != nil {
+		return "", err
+	}
+
+	// Generate the desired file path
+	desiredFilePath := filepath.Join(os.TempDir(), fileName)
+
+	// Create a temporary CSV file
+	tempFile, err := os.Create(desiredFilePath)
+	if err != nil {
+		return "", err
+	}
+	defer tempFile.Close()
+
+	// Create a CSV writer
+	writer := csv.NewWriter(tempFile)
+
+	// Write header to the CSV file
+	if err := writer.Write(fieldNames); err != nil {
+		return "", err
+	}
+
+	// Write records to the CSV file
+	for _, record := range records {
+		var row []string
+		fields, ok := record["message"].(map[string]interface{})["fields"].(map[string]interface{})
+		if !ok {
+			return "", fmt.Errorf("fields not found in a record")
+		}
+		for _, key := range fieldNames {
+			row = append(row, fmt.Sprintf("%v", fields[key]))
+		}
+		if err := writer.Write(row); err != nil {
+			return "", err
+		}
+	}
+
+	// Flush any buffered data to the underlying writer
+	writer.Flush()
+	if err := writer.Error(); err != nil {
+		return "", err
+	}
+
+	// Get the absolute path of the temporary file
+	absPath, err := filepath.Abs(tempFile.Name())
+	if err != nil {
+		return "", err
+	}
+
+	return absPath, nil
+}
+
+func getFileFormat(path string) (string, error) {
+	extension := filepath.Ext(path)
+	if extension == "" {
+		return "", errors.New("file extension not found")
+	}
+	return extension, nil
+}
+
+func getUploadFilePath(path string) (string, error) {
```

Comment:
did you check if any function is available for this?

Reply:
No, no existing utility is available similar to above function